src/ast/codegen_llvm.cpp

```diff
@@ -834,6 +834,28 @@ void CodegenLLVM::visit(Call &call)
     b_.CreateLifetimeEnd(buf);
     expr_ = nullptr;
   }
+  else if (call.func == "strftime")
+  {
+    auto elements = AsyncEvent::Strftime().asLLVMType(b_);
+    StructType *strftime_struct = b_.GetStructType(call.func + "_t",
+                                                   elements,
+                                                   true);
+
+    AllocaInst *buf = b_.CreateAllocaBPF(strftime_struct, call.func + "_t");
+    b_.CreateStore(b_.GetIntSameSize(asyncactionint(AsyncAction::strftime),
+                                     elements.at(0)),
+                   b_.CreateGEP(buf, { b_.getInt64(0), b_.getInt32(0) }));
+
+    b_.CreateStore(b_.GetIntSameSize(time_id_, elements.at(1)),
+                   b_.CreateGEP(buf, { b_.getInt64(0), b_.getInt32(1) }));
+    strftime_id_++;
+    Expression &arg = *call.vargs->at(1);
+    arg.accept(*this);
+    b_.CreateStore(expr_,
+                   b_.CreateGEP(buf, { b_.getInt64(0), b_.getInt32(2) }));
+    expr_ = buf;
+    // b_.CreateLifetimeEnd(buf);
```

Comment:
You'll want to put this in `expr_deleter_`: https://github.com/iovisor/bpftrace/blob/master/src/ast/codegen_llvm.h#L91 .

Callers should be running `expr_deleter_`. My #1386 should make it harder to forget about calling the deleter.